bofire/data_models/strategies/predictives/enting.py

```diff
@@ -0,0 +1,89 @@
+from typing import Any, Dict, Literal, Type
+
+from pydantic import Field, PositiveFloat, PositiveInt
+
+from bofire.data_models.base import BaseModel
+from bofire.data_models.constraints.api import (
+    Constraint,
+    LinearEqualityConstraint,
+    LinearInequalityConstraint,
+    NChooseKConstraint,
+)
+from bofire.data_models.features.api import (
+    CategoricalDescriptorInput,
+    CategoricalInput,
+    ContinuousInput,
+    ContinuousOutput,
+    DiscreteInput,
+    Feature,
+)
+from bofire.data_models.objectives.api import (
+    MaximizeObjective,
+    MinimizeObjective,
+    Objective,
+)
+from bofire.data_models.strategies.predictives.predictive import PredictiveStrategy
+
+
+class UncParams(BaseModel):
+    beta: PositiveFloat = 1.96
+    bound_coeff: float = 0.5
+    acq_sense: Literal["exploration", "penalty"] = "exploration"
+    dist_trafo: Literal["normal", "standard"] = "normal"
+    dist_metric: Literal["euclidean_squared", "l1", "l2"] = "euclidean_squared"
+    cat_metric: Literal["overlap", "of", "goodall4"] = "overlap"
+
+
+class TrainParams(BaseModel):
+    # lightgbm training hyperparameters
+    objective: str = "regression"
+    metric: str = "rmse"
+    boosting: str = "gbdt"
+    num_boost_round: PositiveInt = 100
+    max_depth: PositiveInt = 3
+    min_data_in_leaf: PositiveInt = 1
+    min_data_per_group: PositiveInt = 1
+    verbose: int = -1
+
+
+class TreeTrainParams(BaseModel):
+    train_params: "TrainParams" = Field(default_factory=lambda: TrainParams())
+    train_lib: Literal["lgbm"] = "lgbm"
```

Comment:
Hmm, ok. I understand. Then proceed as you wish. @bertiqwerty @R-M-Lee what do you think?

Reply:
I would make the life for BoFire user as easy as possible. Creating the entmoot-lib-parameters where needed in the corresponding code during instantiation is fine, even though their structure is different. We do not need to replicate this here, if there is no benefit for the BoFire user.

Reply:
So you would also argue for a flat data structure in the strategy and the corresponding data structures for `entmoot` are generated on the fly based on the flat structure and passed to the corresponding function calls? This would be also my preferred solution, as this makes it easy for the bofire user. Thing like the parameter `regression` for LGBM can then also removed, as this is generated under the hood before passed to LGBM. 

I think also that it can tested quite well. Just create a helper method that takes the flat arguments in and that returns the data structures needed for entmoot, which then can be easilty tested.

Reply:
I think this all makes sense. I have created two methods in the `datamodels.EntingStrategy` class that dump the flat structure into the nested structure required for ENTMOOT. This also means that I have removed the unecessary parameters. I hope this solution is what you were thinking!